weblab/entities/views.py

```diff
@@ -778,6 +779,42 @@ def get(self, request, *args, **kwargs):
         return response
 
 
+class RenameView(LoginRequiredMixin, UserPassesTestMixin, FormMixin, EntityTypeMixin, DetailView):
+    template_name = 'entities/entity_rename_form.html'
+    context_object_name = 'entity'
+    form_class = EntityRenameForm
```

Comment:
Similarly to `EntityCreateView`, you can have 2 subclasses of `EntityRenameForm`, one for models and one for protocols (and a third for `FittingSpec` in the `fitting` app actually). And rather than defining `form_class` like this, make it an `@property` which returns the form class corresponding to the class `self.model`.